### ISSUE TITLE
Provide more informative setup instructions

### ISSUE THREAD

**Issue by renkun-ken** [Ikuyadeu/vscode-R#241]
As I commented at https://github.com/Ikuyadeu/vscode-R/issues/240#issuecomment-590014087:

>We may need a major improvement of README to make clear the setup of multiple parts including vscode-R, vscode-r-lsp, languageserver, and radian on Windows, Linux/macOS so that it is easier for user to have the full experience these parts combined has to offer.

I suggest we write more detailed setup instructions in README or in wikis.

**Comment by andycraig** [comment]
I agree. A couple of thoughts:

1. I think we should suggest that users install radian as part of setup, i.e., have radian be vscode-r’s ‘recommended console’
2. On vscode-R activation, we could check if vscode-r-lsp is installed, and prompt for installation if it isn’t
3. In vscode-r-lsp, we could check if languageserver is installed (I think this is possible), and prompt for installation if it isn’t

**Comment by renkun-ken** [comment]
I'm also wondering if we have a way to find R binary automatically on Windows? In my knowledge, most beginners don't know where the R binary is located or can be confused by `x86`, `x64` stuff.

**Comment by renkun-ken** [comment]
I suggest we create a wiki to present more documentation:

* Getting started
* Installation
  * Windows
  * Linux/macOS
* R Terminal
* Keyboard shortcuts
* R Session watcher
* Recommended extensions and configuration
  (such as C/C++ to write Rcpp, Path Autocomplete, TODO highlight, etc. mainly for RStudio user to find similar editor features in VSCode)

**Comment by renkun-ken** [comment]
I created several wiki pages at https://github.com/Ikuyadeu/vscode-R/wiki. I'll spend some time to finish these pages later.

**Comment by renkun-ken** [comment]
I've finished writing some introduction pages and installation guides. Will elaborate on other pages.

**Comment by Ikuyadeu** [comment]
Great works!
Also, I put the contribution guide from `CONTRIBUTING.md`. It can be updated by just putting the wiki link too.

**Comment by phillc73** [comment]
I'm pretty new to vscode and am looking to migrate from RStudio. Therefore, some of my comments might be a bit too basic and self-evident, so I apologise for that.

I think there are still some aspects of the setup missing, and maybe they're more to do with understanding vscode, than specifically vscode-R.

I'm missing some simple steps to follow regarding setting up a workspace environment to launch just the way I like it. I have a setup/layout I like to use, but I seem to somehow default back to just one large pane view whenever I open a new folder, which I then need to split into the multi-pane view I prefer. Also, first thing I do when I start vscode now is hit F1 and select Attach Active Terminal. There must be something I've missed in the setup instructions for this to happen automatically on startup.

Or something like the equivalent of RStudio's Project scaffolding to create a default Workspace view/layout.

Basically I'm looking for the most friction free way for a user to migrate to vscode from RStudio. It could be a comparison table perhaps of common options from RStudio, and how to do the equivalent in vscode. e.g. from RStudio:

* New Project (Workspace) with default layout
* New R Package
* New R markdown document
* etc

**Comment by renkun-ken** [comment]
@phillc73 Thanks for your kind suggestions!

I was just thinking about a feature table of RStudio and VSCode R language support. I'll write something like this when I have time.

**Comment by D3SL** [comment]
Some more documentation on features like session watcher and terminals would be helpful as well. As it is written now it sounds like just by adding ```source(file.path(Sys.getenv(if (.Platform$OS.type == "windows") "USERPROFILE" else "HOME"), ".vscode-R", "init.R"))
``` to my Rprofile I'll be able to automagically use VSCode with any R or radian session I spawn, be it out of cmd or via VSCodes "Start external terminal" option, and have it work just like the "R: Create R Terminal" option in terms of integration and being able to use shortcuts like ctrl+enter.

I think for a lot of RStudio converts like me one of the major drives behind sticking with it so long is the way it "GUIifies" R for us with the packages and environment panes. Especially when it comes to things like stepping through a function and seeing every object update live.

**Comment by renkun-ken** [comment]
Closing as resolved.
